addons/portal/controllers/thread.py

```diff
@@ -24,3 +25,19 @@ def _can_edit_message(cls, message, hash=None, pid=None, token=None, **kwargs):
             if partner and message.author_id == partner:
                 return True
         return super()._can_edit_message(message, hash=hash, pid=pid, token=token, **kwargs)
+
+    def _get_fetch_domain(self, thread, *args, fetch_portal_message=None, **kwargs):
+        """Restricts the fetched messages for portal."""
+        domain = super()._get_fetch_domain(
+            thread, *args, fetch_portal_message=fetch_portal_message, **kwargs
+        )
+        model = request.env[thread._name]
+        if fetch_portal_message:
+            domain = Domain.AND(
+                [
+                    domain,
+                    request.env["mail.message"]._get_search_domain_share(),
```

Comment:
as discussed: this `_get_search_domain_share` is repeated, should have a condition to determine if it should be there or not (only once)